tools/autograd/gen_annotated_fn_args.py

```diff
@@ -28,7 +28,7 @@
     is_py_nn_function, is_py_linalg_function, is_py_variable_method
 
 def gen_annotated(native_yaml_path: str, out: str, autograd_dir: str) -> None:
-    native_functions = parse_native_yaml(native_yaml_path)
+    native_functions, _ = parse_native_yaml(native_yaml_path)
```

Comment:
Consider returning a named tuple here

Reply:
then `parse_native_yaml(native_yaml_path).native_functions`

Reply:
(Another alternative, though no point in changing it now, would have been to keep `parse_native_yaml` as its original meaning, and then add a new function which also returns the tuple. Since you already pushed the changed through no further action necessary).

Reply:
Apropos of nothing, all of this repeated parsing smells like an antipattern

Reply:
Sounds good. I'm also updating `parse_backend_yaml` to return a named tuple, it seems like a minor readability win.

> all of this repeated parsing smells like an antipattern

Maybe the codegen should all be called from a single entry point, rather than being a bundle of python scripts that the build systems needs to call separately? I'm guessing the current setup is just a relic of the fact that autograd/tracing codegen used to read in `Declarations.yaml`, which was an output of `gen.py`.

Reply:
> Maybe the codegen should all be called from a single entry point, rather than being a bundle of python scripts that the build systems needs to call separately? I'm guessing the current setup is just a relic of the fact that autograd/tracing codegen used to read in Declarations.yaml, which was an output of gen.py.

Yep. Not for this PR though!